### ISSUE TITLE
Problems preventing Travis from building PRs

### ISSUE THREAD

**Issue by edunham** [travis-ci/travis-ci#4145]
EDIT: I tried to do things through the config file which are only available for the web UI. Solution is to go to 
https://travis-ci.org/owner/repor/settings 
and click the pretty button beside "build pull requests".

-------------------------------------------------------------------------------------------------------

The problem is probably with either the documentation or the branch restriction commands themselves. 

#### Expected behavior: 

* I Google for "Travis skip building PR" or "Travis build only on push to master" and find a section of Travis docs telling me how to prevent builds from being triggered by pull requests. 
* Or documentation around `branches.only` explicitly states whether the `branches.only` setting is meant to dissable building on pull requests, and directs the user to how to do that. 
* Or documentation for `before_deploy` and related commands explicitly states whether or not those steps will be run when Travis builds a PR
* Or the docs on [deploying to S3](http://docs.travis-ci.com/user/deployment/s3/) discuss the problem of building PRs. 
* Or the docs on [testing pull requests](http://docs.travis-ci.com/user/pull-requests/) say how to prevent building PRs.

#### Actual behavior: 

What I find when trying to build only on commits to master (and never build PRs) is an [inconclusive closed issue](https://github.com/travis-ci/travis-ci/issues/1316) and [the related Travis docs](http://docs.travis-ci.com/user/customizing-the-build/#Building-Specific-Branches). The docs do not specify which commands will be run when Travis runs on PRs and which will only be run on commits to master. 

#### An Example:

I followed the [deploying to S3](http://docs.travis-ci.com/user/deployment/s3/) docs to write [this .travis.yml](https://github.com/rust-lang/rust-www/blob/master/.travis.yml). Note that I tried to tell it to build only on master with

```
branches:
  only:
  - master
```

However, when I make [a pull request](https://github.com/rust-lang/rust-www/pull/154) to the repo, Travis runs on the PR (and obviously fails, since it [can't decrypt the keys](http://docs.travis-ci.com/user/pull-requests/) for very good security reasons). 

The only reason I'm using Travis on this repository at all is to build and deploy the site when PRs are merged into the master branch. These are not actions that should ever be taken at the time a PR is first opened. 

If somebody could explain what the correct technique is for building only after a PR is merged, I'll propose an update to the docs that would have prevented me from filing this ticket. 

Thanks!
